### PR TITLE
feat: import features and fixes from pisp/master

### PR DESCRIPTION
**Pull request by kleyow**
Couldn't PR pisp/master into master nicely so I've just imported the changes :|

### CODE REVIEW

**Comment by kleyow** [comment]
@lewisdaly I think here as well there might not be a need for ENV vars for stuff like
- FEATURE_ENABLE_DETAILED_ACCOUNT_LOOKUP
- FEATURE_ENABLE_EXTENDED_PARTY_ID_TYPE

The changes shouldn't break anything and are simply validation/payload schema changes.
`accounts` field is opitional and the new enums aren't really a big change.

Lmk if it's worth doing the feature switches for these.

**Comment by lewisdaly** [comment]
I'm fine with leaving the accounts stuff in for now since, as you say, it's optional - plus we will be migrating that logic to the 3p-API-Adapter soon.

For the additional party types, I worry that we are altering functionality as it is defined in the FSPIOP-API, which is something we shouldn't be doing without changing the API. Perhaps if there's no way to easily switch out the Enum definitions with an environment variable, you could add a manual validation step before any handler that checks for the `PartyIdType`, and if it's in the "extended" list, but the `FEATURE_ENABLE_EXTENDED_PARTY_ID_TYPE` flag is false, then throws a validation error. That way we aren't introducing new behaviour unintentionally.

**Comment by kleyow** [comment]
@lewisdaly Hmm...I'm not sure how to middleman hapi's lifecycle like that. 

So I've gone with my initial idea of a config value that switches between the extended definition and the fspiop api thats already define.
Hope that is an ok stop gap for now? Take a look. Please and thank you.

**Comment by lewisdaly** [comment]
Yeah that was my preferred approach anyway

File: src/server.js [hunk 1]
```diff
@@ -115,7 +121,13 @@ const initializeApi = async (port = Config.API_PORT) => {
...
   await connectDatabase()
   await migrate()
-  const OpenAPISpecPath = Path.resolve(__dirname, './interface/admin-swagger.yaml')
+  let apiPath
+  if (Config.FEATURE_ENABLE_EXTENDED_PARTY_ID_TYPE) {
+    apiPath = './interface/thirdparty/admin-swagger.yaml'
+  } else {
+    apiPath = './interface/admin-swagger.yaml'
+  }
+  const OpenAPISpecPath = Path.resolve(__dirname, apiPath)
```
Comment (lewisdaly) [inline src/server.js:130]: Looks good, although I'd probably generalize this into it's own function... maybe something like:

```js
const OpenAPISpecPath = pathForConfig(Config)
```

`pathForConfig` can check: (1) is this the admin or non-admin API? and (2) should we enable the extended party id type?

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: src/server.js (hunk 1 above)
Comment (kleyow) [reply src/server.js]: done.

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**